src/shape.hpp

```diff
@@ -67,6 +67,11 @@ class Dimension {
     bool match(dimension_value_t value) const;
     bool partiallyFitsInto(const Dimension& value) const;
     bool isAny() const;
+    std::optional<Dimension> createIntersection(const Dimension& other) const;
+
+private:
+    dimension_value_t getMinValueRegardless() const;
```

Comment:
Maybe use `getLowerBound()` and `getHigherBound` naming?

Reply:
Good idea